src/run_tribler.py

```diff
@@ -57,17 +62,44 @@ def main() -> None:
     root_state_dir = get_root_state_directory(os.environ.get('TSTATEDIR', 'state_directory'))
     logger.info("Root state dir: %s", root_state_dir)
 
-    api_port, api_key = int(os.environ.get('CORE_API_PORT', '-1')), os.environ.get('CORE_API_KEY')
+    api_port, api_key = int(os.environ.get('CORE_API_PORT', '0')), os.environ.get('CORE_API_KEY')
+
+    config = TriblerConfigManager(root_state_dir / "configuration.json")
+    config.set("state_dir", str(root_state_dir))
+
+    if config.get("api/refresh_port_on_start"):
+        config.set("api/http_port", 0)
+        config.set("api/https_port", 0)
+
+    if api_key is None and config.get("api/key") is None:
+        api_key = os.urandom(16).hex()
+
+    if api_key is not None and api_key != config.get("api/key"):
+        config.set("api/key", api_key)
+        config.write()
+
+    if api_port is not None and api_port != config.get("api/http_port"):
+        config.set("api/http_port", api_port)
+        config.write()
+
+    logger.info("Start tribler core. API port: %d. API key: %s.", api_port, config.get("api/key"))
+
+    session = Session(config)
+    await session.start()
+
+    image_path = Path(__file__).absolute() / "../tribler/ui/public/tribler.png"
+    image = Image.open(image_path.resolve())
+    url = f"http://localhost:{session.rest_manager.get_api_port()}/ui/#/downloads/all?key={config.get('api/key')}"
+    menu = (pystray.MenuItem('Open', lambda: webbrowser.open_new_tab(url)),
+            pystray.MenuItem('Quit', lambda: session.shutdown_event.set()))
+    icon = pystray.Icon("Tribler", icon=image, title="Tribler", menu=menu)
+    threading.Thread(target=icon.run).start()
 
-    # Check whether we need to start the core or the user interface
-    if parsed_args["core"]:
-        from tribler.core.start_core import run_core
-        run_core(api_port, api_key, root_state_dir)
-    else:
-        # GUI
-        from tribler.gui.start_gui import run_gui
-        run_gui(api_port, api_key, root_state_dir)
+    await session.shutdown_event.wait()
+    await session.shutdown()
+    icon.stop()
+    logger.info("Tribler shutdown completed")
 
 
 if __name__ == "__main__":
-    main()
+    asyncio.run(main())
```

Comment:
Won't this ruin the `asyncio.set_event_loop(asyncio.SelectorEventLoop())` on Windows?

Reply:
You're right. I'll move the `set_event_loop` call to just before calling `run`.